lutris/util/timer.py

```diff
@@ -2,20 +2,30 @@
 
 
 class Timer:
+    """Simple Timer class to time code.
+    You can overload the + operator according to your need by using a new timer_type.
+    """
+
+    def __init__(self, timer_type=None):
+        self.timer_type = timer_type
+        self.finsihed = False
 
     def start_t(self):
-        self.start = datetime.datetime.now()
+        self._start = datetime.datetime.now()
 
     def end_t(self):
-        self.end = datetime.datetime.now()
+        self._end = datetime.datetime.now()
+        self.finsihed = True
 
     def duration(self):
-        return (self.end - self.start).seconds / 3600
+        return (self._end - self._start).seconds / 3600
 
-    def increment(self, saved_dur):
-        if saved_dur == '':
-            return "%.1f hrs" % self.duration()
+    def __add__(self, saved_dur):
```

Comment:
\+ is used for addition or concatenation in Python. This method does neither really. While it does addition, it also converts the number to a string and does formatting. Also \_\_add\_\_ shouldn't return None rather raise an exception.

\_\_add\_\_ should probably accept either a number type or timer instance.

Having said that when I proposed the usage of dunder methods I had \_\_str\_\_ in mind for formatting purposes.